src/main/java/bio/terra/service/dataset/flight/LockDatasetStep.java

```diff
@@ -58,7 +59,7 @@ public StepResult doStep(FlightContext context) {
       } else {
         return new StepResult(StepStatus.STEP_RESULT_FAILURE_FATAL, notFoundEx);
       }
-    } catch (RetryQueryException | DatasetLockException e) {
+    } catch (RetryQueryException | DatasetLockException | TransactionSystemException e) {
```

Comment:
Why are the retryable exceptions in the undo method different from the retryable ones in the do method (and in the unlock step's do method)?

Reply:
Ah derp, you're totally right.  I'll update in a followon